igvm/vm.py

```diff
@@ -558,25 +622,25 @@ def aws_build(self,
         :raises: VMError: Generic exception for VM errors of all kinds
         """
 
-        ec2 = self.aws_session.client('ec2')
-
         vm_types_overview = self.aws_get_instances_overview()
         if vm_types_overview:
             vm_types = self.aws_get_fitting_vm_types(vm_types_overview)
         else:
             vm_types = [self.dataset_obj['aws_instance_type']]
 
-        ec2_res = self.aws_session.resource('ec2')
         root_device = list(
-            ec2_res.images.filter(
+            self.ec2r.images.filter(
                 ImageIds=[self.dataset_obj['aws_image_id']]
             )
         )[0].root_device_name
         disk_size_gib = self.dataset_obj['disk_size_gib']
 
+        for aws_sg in self.aws_sgs:
+            self.dataset_obj['aws_security_group_ids'].add(aws_sg.group_id)
```

Comment:
@lacunoc Should we rather replace manually assigned groups instead of appending to them?

Reply:
Yes, we should.

Reply:
The attribute is now still populated and commited but overwritten. Removal of attributes will be in another MR.